### ISSUE TITLE
render_ahead_of_time/render_ahead_safezone per monitor

### ISSUE THREAD

**Issue by nishiiko** [hyprwm/Hyprland#8892]
### Description

render_ahead_of_time can help get low latency without tearing
however, when using lower values, it can cause issues on setups with mismatching refresh rates
e.g. in my case, my main monitor (144hz) is best with a safezone of 6ms, but this will leave my second monitor (75hz) basically unusable because of the low safezone value
the idea is to be able to set different values for each monitor, for example, 6ms safezone for DP-1, and 12ms for HDMI-A-1

is this possible? or does the way render_ahead_of_time works prevent this idea

**Comment by Fxzzi** [comment]
is render ahead of time (RAT) actually useful if you also have tearing enabled? I would need vaxry's input on this but I think RAT is useless with tearing. Soooo, enable tearing for your games if you need the lowest latency

**Comment by vaxerski** [comment]
RAT is to be removed. It makes little sense to me, since we have tearing.

**Comment by nishiiko** [comment]
TLDR: RAT gives low latency even with notifications and overlays popping up, tearing disables when ntofications and overlahys pop up and can make input delay feel awkward as it becomes so varied.
-------------------------------------
i think there is still a usecase for it because with how tearing currently functions, any notifications or popups that might come up will disable it and your latency will fluctuate quite a bit. RAT lets you keep your low latency even while other notifications and overlays show up on your screen
along with that, having no tearing with low latency makes things a little more clearer ingame. admittedly, not extremely significantly, but to a noticeable degree at least to me
i have notifications popping up quite often in situations where, for example, i am talking to a friend in vc where they cant talk for a while and they have to type on discord instead. it lets me read their notifications while playing without having to switch workspaces or crane my head over to my second monitor constantly.
if RAT were to be removed, i would like to see an option where tearing could be enabled at all times, or forced when a fullscreen application is detected
if none of that is possible with current implementations then... damn, ill just deal with it TvT

**Comment by vaxerski** [comment]
tearing itself doesn't require that nothing is drawn over it, it's just how hyprland does it because it's simpler code-wise

**Comment by vaxerski** [comment]
> RAT lets you keep your low latency even while other notifications and overlays show up on your screen
along with that, having no tearing with low latency makes things a little more clearer ingame. admittedly, not extremely significantly, but to a noticeable degree at least to me

Honestly in my mind if you want low latency on 60hz screens you are already a bit insane, especially since 144/240hz panels are dirt cheap nowadays.

**Comment by nishiiko** [comment]
> tearing itself doesn't require that nothing is drawn over it, it's just how hyprland does it because it's simpler code-wise

is it planned later on to allow tearing while other things are being drawn?
